content/en/docs/concepts/configuration/secret.md

```diff
@@ -15,48 +15,370 @@ weight: 30
+merged into the `data` field. If a key appears in both the `data` and the
+`stringData` field, the value specified in the `stringData` field takes
+precedence.
+
+### Secret Types
+
+When creating a Secret, you can specify its type using the `type` field of
+the [`Secret`](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#secret-v1-core)
+resource, or certain equivalent `kubectl` command line flags (if available).
+The Secret type is used to facilitate programmatic handling of the Secret data.
+
+Kubernetes provides several builtin types for some common usage scenarios.
+These types vary in terms of the validations performed and the constraints
+Kubernetes imposes on them.
+
+| Builtin Type | Usage |
+|--------------|-------|
+| `Opaque`     |  arbitrary user-defined data |
+| `kubernetes.io/service-account-token` | service account token |
+| `kubernetes.io/dockercfg` | serialized `~/.dockercfg` file |
+| `kubernetes.io/dockerconfigjson` | serialized `~/.docker/config.json` file |
+| `kubernetes.io/basic-auth` | credentials for basic authentication |
+| `kubernetes.io/ssh-auth` | credentials for SSH authentication |
+| `kubernetes.io/tls` | data for a TLS client or server |
+| `bootstrap.kubernetes.io/token` | bootstrap token data |
+
+You can define and use your own Secret type by assigning non-empty string as the
+`type` value for a Secret object. An empty string is treated as `Opaque`.
+Kubernetes doesn't impose any contstraints on the type name you choose. But if
+you are using one of the builtin types listed above, you will have to meet all
+the requirements defined for that type.
+
+#### Opaque Secrets
+
+`Opaque` is the default Secret type if omitted from a Secret configuration file.
+When you create a Secret from the `kubectl` command line, it corresponds to
+the `generic` subcommand. For example, the following command creates an empty
+Secret of type `Opaque`.
 
-### Built-in Secrets
+```shell
+kubectl create secret generic empty-secret
+kubectl get secret empty-secret
+```
 
-#### Service accounts automatically create and attach Secrets with API credentials
+The output looks like:
+
+```
+NAME           TYPE     DATA   AGE
+empty-secret   Opaque   0      2m6s
+```
+
+The `DATA` column shows the number of data items stored in the Secret.
+In this case, `0` means we have just created an empty Secret.
+
+####  Service account token Secrets
+
+A `kubernetes.io/service-account-token` type of Secret is used to store a
+token that identifies a service acccount. When using this Secret type, you need
+to ensure that the `kubernetes.io/service-account.name` annotation is set to an
+existing service account name. An Kubernetes controller will fill in some other
+fields such as the `kubernetes.io/service-account.uid` annotation and the
+`token` key in the `data` field set to actual token content.
+
+The following YAML is an example config for a service account token Secret:
```

Comment:
nit: The following example configuration declares a service account token Secret: